### PR TITLE
JS: Fix `import {a as a}` and `export {a as a}` format

### LINKED ISSUE #9434: Prettier should not convert `export {a as a}` to `export {a}`
**Prettier pr-9408**
[Playground link](https://deploy-preview-9408--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACYAQz0IGcSBfPAM2wgFs8ByQpkAGhAgxgEtpSyUIWx0A7gAURCQSkIAbMYQCegzgCNshMAGs4MAMoZtvKAHNkMbAFc4nOPXVwAJs5cAZQueuEzcAGI49IQwfObIIITWMBAcIAAWMPTyAOrxvPCkxmBwBjIZvABuGcoRYKRqIKakcLgSWmbByNQKNZwAVqRoAEJauvoGhPRw7qZwza12IJ1oBqZm8nAAitYQ8BPybSDG2DXYEeqETvJxGNimMCm8zjDxyAAcAAycZxA1KVoYEWdwe4XjnAAjqt4PVuLJIqQALRQOAuFxxbBwYG8JH1XxNJAtTZTGr0XiWGy4+aLFZrcZYyacGBHK43O5IABM1K0vHk8wAwgxMahSABWOLWGoAFSOsmxW0KtgAklA3LADGBzjwAIJygwwZSLDY1CgUIA)
```sh
--parser babel
```

**Input:**
```jsx
export {a as a} from 'a'
```

**Output:**
```jsx
export { a } from "a";

```

**Expected behavior:**
```jsx
export { a } from "a";

```

And comment will lost

**Prettier pr-9408**
[Playground link](https://deploy-preview-9408--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACYAQz0IGc8B6AKj0gFs6F8qKSBfPAM2wjrwHJC-EABoQEDDACW0UslCFsPAO4AFRQjkpCAG2WEAnnLEAjbITABrODADKGC1KgBzZDGwBXOGLh0TcABMAwIAZQhcPQmc4ADEcOkIYaRdkEEIPGAhREAALGDodAHUcqXhSBzA4W01SqQA3UoNUsFJjECdSOFxVc2cE5E5dTrEAK1I0ACFzKxtbQkYQpzgBoe8QMbRbJ2cdOABFDwh4FZ1hkAdsTuxUk0J-HWyMbCcYQqkAmBzkAA4ABjEnhBOoVzBhUk84Fc6ssxABHQ7wHoSLRpUgAWigcECgWy2Dg8KkeJ6UX6SEGpzWnToUjcnkp212ByOyzJqzEMDubw+XyQACZ2eYpDptgBhXik1CkACs2Q8nQAKnctOSznUvABJKDBWC2MDPSQAQS1thgBl2J06bDYQA)
```sh
--parser babel
```

**Input:**
```jsx
export {a as /* comment */ a} from 'a'
```

**Output:**
```jsx
Error: Comment "comment" was not printed. Please report this error!
    at https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:15:111760
    at Array.forEach (<anonymous>)
    at Object.ensureAllCommentsPrinted (https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:15:111727)
    at Xi (https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:15:117453)
    at Yi (https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:15:119639)
    at formatWithCursor (https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:15:119759)
    at https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:55:37207
    at Object.format (https://deploy-preview-9408--prettier.netlify.app/lib/standalone.js:55:37290)
    at formatCode (https://deploy-preview-9408--prettier.netlify.app/worker.js:115:21)
    at handleMessage (https://deploy-preview-9408--prettier.netlify.app/worker.js:66:18)
```

### PR DESCRIPTION
**Pull request by fisker**
Fixes #9434

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Review by kasunsewwandika — APPROVED** [review]

**Review by sosukesuzuki — APPROVED** [review]

**Comment by kachkaev** [comment]
Is it possible to keep `as` only where there are extra nodes in the AST such as comments? What’s the downside of converting plain `{ a as a }` into `{ a }`?

**Comment by sosukesuzuki** [comment]
I think it is a what Prettier should not do. It's a linter's job. Prettier should not break AST at least.

**Comment by fisker** [comment]
@kachkaev Another motivation https://github.com/prettier/prettier/pull/9408#discussion_r507405239, we don't want make 

```js
export { "prettier" as 'prettier' };
export { "prettier" as "prettier" };
export { "pr\u{0065}ttier" as "prettier" };
```

any different, and the first one will be unstable.

The following are even more complicated

```js
export { "pret\'tier" as 'prettier' };
export { "pret'tier" as 'prettier' };
export { 'pret\'tier' as 'prettier' };
```

File: src/language-js/utils.js [hunk 1]
```diff
@@ -342,8 +342,28 @@ function isGetterOrSetter(node) {
...
+function sameLocEnd(nodeA, nodeB, { locEnd }) {
+  return locEnd(nodeA) === locEnd(nodeB);
+}
+
+/**
+ * @param {Node} nodeA
+ * @param {Node} nodeB
+ * @returns {boolean}
+ */
+function isSameNode(nodeA, nodeB, options) {
```
Comment (thorn0) [inline src/language-js/utils.js:363]: `isSameLoc` is a better name. Different nodes can have the same location. E.g.
```json
      {
        "type": "ExpressionStatement",
        "start": 0,
        "end": 3,
        "loc": {
          "start": { "line": 1, "column": 0 },
          "end": { "line": 1, "column": 3 }
        },
        "range": [0, 3],
        "expression": {
          "type": "Identifier",
          "start": 0,
          "end": 3,
          "loc": {
            "start": { "line": 1, "column": 0 },
            "end": { "line": 1, "column": 3 },
            "identifierName": "fff"
          },
          "range": [0, 3],
          "name": "fff"
        }
      }
```

**Review by evilebottnawi — APPROVED** [review]

File: src/language-js/utils.js (hunk 1 above)
Comment (fisker) [reply src/language-js/utils.js]: Renamed as `hasSameLoc` https://github.com/prettier/prettier/pull/9435/commits/0d515b93a89b51b1b70bc4e31a663bc0caf75156

**Review by thorn0 — APPROVED** [review]

**Pull request merged**